site/content/integrate/plugins/components/webapp/best-practices.md

```diff
@@ -24,6 +24,7 @@ You can additionally [register a slash command]({{< ref "/integrate/plugins/comp
 ### Actions that apply to specific messages
 - Recommendation: Have your plugin register a [post dropdown menu component]({{< ref "/integrate/plugins/components/webapp/reference#registerPostDropdownMenuComponent" >}}) with some text, icon and an action function. This adds your action to the "More Actions" post menu dropdown for easy discovery.
 - Examples: Create or attach to Jira issue from a message; copy a message to another channel; report an inappropriate message
+- Sample code: [mattermost/mattermost-plugin-demo/webapp/src/plugin.jsx](https://github.com/mattermost/mattermost-plugin-demo/blob/master/webapp/src/plugin.jsx#L119)
```

Comment:
The demo plugin example actually may not be the best example to recommend here (my apologies as I recommended this 🙂), since it's not actually using the post id from the post that the user is interacting with. The example from the Todo plugin may be more appropriate, since it shows how to access the `postId` of the post that the user is interacting with, whereas the demo plugin is a more basic implementation that doesn't take into account the post itself.

```ts
registry.registerPostDropdownMenuAction(
    'Add Todo',
    (postID) => {
        telemetry('post_action_click');
        store.dispatch(openAddCard(postID));
        store.dispatch(showRHSPlugin);
    },
);
```

```suggestion
- Sample code: [mattermost/mattermost-plugin-todo/webapp/src/index.js](https://github.com/mattermost/mattermost-plugin-todo/blob/0c4dbfb58a72f8392ea66e101996afd06fdb2913/webapp/src/index.js#L30-L37)
```